### PR TITLE
Refactor title case lib

### PR DESCRIPTION
**Pull request by afragen**
Removed nested ternary equations.

Jump start for extending `TitleCase`

### CODE REVIEW

**Comment by afragen** [comment]
So, for fun, I split out the _rules_ to `english()`.

I also added a JSON file with a lot of locales and names. No idea if it will be useful.

**Comment by afragen** [comment]
BTW, with these changes I think we've gone into the _fork_ realm and we may want to move these outside the `vendor` directory as it's no longer appropriate.

**Comment by tommcfarlin** [comment]
> BTW, with these changes I think we've gone into the fork realm and we may want to move these outside the vendor directory as it's no longer appropriate.

I agree. And in forking it, we'll obviously continue to give attribution but this will also give us the freedom to bring the coding standards on par for WPCS and with PHP-FIG. 

IIRC, @GaryJones tends to lean in the direction of PHP-FIG and I tend to lean more in terms of WPCS (since this is a WPCS-specific plugin). To that end, that's the direction and the ruleset I'll be using.

I'm always open to hearing reasons for using a different coding standards in some respects than what WordPress recommends (and I do wish WPCS was a bit more up to speed with the PHP-FIG) but I'm mentioning this now so it's documented somewhere so we have the rationale why I may/may not leave code/merge code in a certain format :).

File: includes/vendor/class-title-case.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ public function toTitleCase( $title ) {
 			$i = mb_strlen( substr( $title, 0, $i ), 'UTF-8' );
 
 			//try to internationalize a bit.
-			$locale       = get_locale();
+			$locale       = get_locale(); //this makes the lib WordPress specific. :(
```
Comment (tommcfarlin) [inline includes/vendor/class-title-case.php]: That's okay with me -- it _is_ a WP-specific plugin :).

**Comment by afragen** [comment]
I will leave the move from the `vendor` directory to `admin` or wherever you see fit up to you. 😉 

I don't know which languages follow similar capitalization rules to English but I suspect most of the romance languages. That means we can add to the `switch` statement those similar languages. I suspect Spanish and French will be similar. German will likely have some variations, though they may be minor. We will need to ask some native speakers for their input.

I tend to lean a little more towards PHP-FIG but I'll go along with the repo owner. It's their project.

**Comment by tommcfarlin** [comment]
> I will leave the move from the vendor directory to admin or wherever you see fit up to you. 

I think it's something that should be moved, for sure, but the thought _has_ crossed by mind that if we're able to break the dependency on WordPress, then we could import it via Composer.

We could then create an intermediary class that would communicate with it so that it still stands alone as its own project but we'd then basically be doing what we're doing now but with a fork of the original project.

This is something for a future milestone, though. 

> That means we can add to the switch statement those similar languages. I suspect Spanish and French will be similar. 

In the short term, I think this will work but using something like a [simple factory](https://en.wikipedia.org/wiki/Factory_(object-oriented_programming)) may ultimately be better. Long `switch` or conditional statements start to become a code smell if they don't have a small set of conditionals, IMHO.

**Comment by GaryJones** [comment]
> To that end, that's the direction and the ruleset I'll be using.

If it's part of a WP plugin, and it's code you've amended even if originally written by others, then yes, it should be WPCS.

> if we're able to break the dependency on WordPress, then we could import it via Composer.

This is exactly what I was thinking. In which case, that new package would be PSR-2.

**Comment by afragen** [comment]
This will be great. I look forward to learning lots along the way. :smile:

**Pull request merged**